### PR TITLE
feat: [CDS-82251]: Add logs for queries that are fetching more than a…

### PR DESCRIPTION
**Pull request by abhinavmittal-harness**
… year data

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/54803)

### CODE REVIEW

File: 120-ng-manager/src/main/java/io/harness/ng/overview/service/CDOverviewDashboardServiceImpl.java [hunk 1]
```diff
@@ -4331,4 +4333,15 @@ private Boolean isK8sOrHelm(
     }
     return Boolean.FALSE;
   }
+
+  public void validateDashboardRequestDuration(long numOfRequestedDays) {
+    if (numOfRequestedDays > 366) {
+      log.warn(String.format(DASHBOARD_DATA_BEING_QUERIED_FOR_MORE_THAN_A_YEAR, numOfRequestedDays));
```
Comment (RishabhGupta34) [inline 120-ng-manager/src/main/java/io/harness/ng/overview/service/CDOverviewDashboardServiceImpl.java:4339]: nit. should we add a `%d` to the format string?

File: 120-ng-manager/src/main/java/io/harness/ng/overview/resource/CDDashboardOverviewResource.java [hunk 2]
```diff
@@ -292,6 +292,7 @@ public ResponseDTO<DashboardWorkloadDeploymentV2> getWorkloadsV2(
     log.info("Getting workloads");
     long numDays = getNumberOfDays(startInterval, endInterval);
     long previousStartInterval = getStartTimeOfPreviousInterval(startInterval, numDays);
+    cdOverviewDashboardService.validateDashboardRequestDuration(numDays);
```
Comment (RishabhGupta34) [inline 120-ng-manager/src/main/java/io/harness/ng/overview/resource/CDDashboardOverviewResource.java:295]: are we only doing this for 1 API?

File: 120-ng-manager/src/main/java/io/harness/ng/overview/service/CDOverviewDashboardServiceImpl.java [hunk 3]
```diff
@@ -4331,4 +4333,15 @@ private Boolean isK8sOrHelm(
...
+    }
+  }
+
+  public void validateDashboardRequestDuration(long startTime, long endTime) {
+    long numberOfDays = (long) Math.ceil((endTime - startTime) / (double) DAY_IN_MS);
```
Comment (sribalij) [inline 120-ng-manager/src/main/java/io/harness/ng/overview/service/CDOverviewDashboardServiceImpl.java:4344]: Can endTime be less than startTime?

File: 120-ng-manager/src/main/java/io/harness/ng/overview/service/CDOverviewDashboardServiceImpl.java [hunk 4]
```diff
@@ -4331,4 +4333,15 @@ private Boolean isK8sOrHelm(
...
+    validateDashboardRequestDuration(numberOfDays);
```
Comment (sribalij) [inline 120-ng-manager/src/main/java/io/harness/ng/overview/service/CDOverviewDashboardServiceImpl.java:4345]: Is it validated in UI? Can they use API ?

**Review by Abhishek-CDC — APPROVED** [review]

File: 120-ng-manager/src/main/java/io/harness/ng/overview/resource/CDDashboardOverviewResource.java (hunk 2 above)
Comment (abhinavmittal-harness) [reply 120-ng-manager/src/main/java/io/harness/ng/overview/resource/CDDashboardOverviewResource.java:295]: Those are done in separate apis

**Review by abhinavmittal-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54803)* status: 3 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/overview/resource/CDDashboardOverviewResource.java` line 295 at r1](https://reviewable.io/reviews/harness/harness-core/54803#gh-1377074789:-Ni3HLOv3lWO25ocg3xz:bb7msri) ([raw file](https://github.com/harness/harness-core/blob/7bf13986c89b52d47e83dd817c7760c774aaf95c/120-ng-manager/src/main/java/io/harness/ng/overview/resource/CDDashboardOverviewResource.java#L295)):*
<details><summary><i>Previously, abhinavmittal-harness (Abhinav Mittal) wrote…</i></summary><blockquote>

Those are done in separate apis
</blockquote></details>

prs*

**Review by abhinavmittal-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54803)* status: 3 file(s) to review. 3. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/overview/service/CDOverviewDashboardServiceImpl.java` line 4344 at r1](https://reviewable.io/reviews/harness/harness-core/54803#gh-1377077970:-Ni3HO5S9KyT00JbbTHX:b-5us4ho) ([raw file](https://github.com/harness/harness-core/blob/7bf13986c89b52d47e83dd817c7760c774aaf95c/120-ng-manager/src/main/java/io/harness/ng/overview/service/CDOverviewDashboardServiceImpl.java#L4344)):*
<details><summary><i>Previously, sribalij (Sridhar) wrote…</i></summary><blockquote>

Can endTime be less than startTime?
</blockquote></details>

That is taken care by FE, also in this case they will not get any data. We are adding logs to understand whether customers are utilising 
___
*[`120-ng-manager/src/main/java/io/harness/ng/overview/service/CDOverviewDashboardServiceImpl.java` line 4345 at r1](https://reviewable.io/reviews/harness/harness-core/54803#gh-1377078157:-Ni3HbcWEPP2wX_YrRwV:b-i1nac1) ([raw file](https://github.com/harness/harness-core/blob/7bf13986c89b52d47e83dd817c7760c774aaf95c/120-ng-manager/src/main/java/io/harness/ng/overview/service/CDOverviewDashboardServiceImpl.java#L4345)):*
<details><summary><i>Previously, sribalij (Sridhar) wrote…</i></summary><blockquote>

Is it validated in UI? Can they use API ?
</blockquote></details>

On ui there is validation on the endtime > startTime but not on the fact that one can't query more than a year data.
This addition of log is to understand that only.

**Review by abhinavmittal-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54803)* status: 3 file(s) to review. 3. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/overview/service/CDOverviewDashboardServiceImpl.java` line 4344 at r1](https://reviewable.io/reviews/harness/harness-core/54803#gh-1377077970:-Ni3I3RF987GOoOncRpm:b-8ce40g) ([raw file](https://github.com/harness/harness-core/blob/7bf13986c89b52d47e83dd817c7760c774aaf95c/120-ng-manager/src/main/java/io/harness/ng/overview/service/CDOverviewDashboardServiceImpl.java#L4344)):*
<details><summary><i>Previously, abhinavmittal-harness (Abhinav Mittal) wrote…</i></summary><blockquote>

That is taken care by FE, also in this case they will not get any data. We are adding logs to understand whether customers are utilising
</blockquote></details>

We want to put restriction on querying data for more than a year but we don't know if people are using it for that purpose cause in that case it could be a breaking change

**Review by RishabhGupta34 — APPROVED** [review]

**Comment by abhinavmittal-harness** [comment]
trigger review

**Comment by abhinavmittal-harness** [comment]
trigger uts

**Comment by abhinavmittal-harness** [comment]
trigger uts

**Comment by abhinavmittal-harness** [comment]
trigger messagecheck

**Pull request merged**